src/web/clients/download/partial.md

```diff
@@ -34,32 +36,29 @@ use reqwest::StatusCode;
 # }
 #
 # impl PartialRangeIter {
-#     pub fn new(content_range: &ContentRangeSpec, buffer_size: u32) -> Result<Self> {
+#     pub fn new(start: u64, end: u64, buffer_size: u32) -> Result<Self> {
 #         if buffer_size == 0 {
 #             Err("invalid buffer_size, give a value greater than zero.")?;
 #         }
 #
-#         match *content_range {
-#             ContentRangeSpec::Bytes { range: Some(range), .. } => Ok(PartialRangeIter {
-#                 start: range.0,
-#                 end: range.1,
-#                 buffer_size,
-#             }),
-#             _ => Err("invalid range specification")?,
-#         }
+#         Ok(PartialRangeIter {
+#             start,
+#             end,
+#             buffer_size,
+#         })
 #     }
 # }
 #
 # impl Iterator for PartialRangeIter {
-#     type Item = Range;
+#     type Item = HeaderValue;
 #
 #     fn next(&mut self) -> Option<Self::Item> {
 #         if self.start > self.end {
 #             None
 #         } else {
 #             let prev_start = self.start;
 #             self.start += std::cmp::min(self.buffer_size as u64, self.end - self.start + 1);
-#             Some(Range::bytes(prev_start, self.start - 1))
+#             Some(HeaderValue::from_str(&format!("bytes={}-{}", prev_start, self.start - 1)).unwrap())

```

Comment:
Can we avoid `unwrap()`?  We are really trying to teach how to implement these functions in production code, to handle runtime errors gracefully.

Reply:
I don't think so. The error case is for things like non-ASCII characters, I suppose. That can be proven statically (`prev_start` and `self.start - 1` are integers, and the format string is fine), but we can't avoid the `unwrap()` until it gets hidden away in the `headers` crate.

Reply:
Added a comment to explain why the `unwrap` never fails.